pkg/internal/integration_test_support/test_server.go

```diff
@@ -0,0 +1,47 @@
+package integrationtestsupport
+
+import (
+	"io"
+	"log/slog"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+
+	"github.com/stretchr/testify/suite"
+	"google.golang.org/protobuf/proto"
+
+	prefabProto "github.com/prefab-cloud/prefab-cloud-go/proto"
+)
+
+func StartTestServer() (*[]*http.Request, *httptest.Server) {
```

Comment:
We post the telemetry to a one-off server so we can have a more real-worldy test